src/machine/state/mem.rs

```diff
@@ -198,3 +198,95 @@ impl IndexMut<OctaAt> for Memory {
         self.buf.index_mut(pos)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn mem_test() {
```

Comment:
It think, you combine more than one test within this function! I suggest to split them!